src/core/server/root/base_path_proxy_root.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { first, k$, toPromise } from '../../lib/kbn_observable';
+
+import { Root } from '.';
+import { DevConfig } from '../dev';
+import { HttpConfig } from '../http';
+import {
+  BasePathProxyServer,
+  BasePathProxyServerOptions,
+} from '../http/base_path_proxy_server';
+
+/**
+ * Top-level entry point to start BasePathProxy server.
+ */
+export class BasePathProxyRoot extends Root {
+  private basePathProxy?: BasePathProxyServer;
+
+  get basePath() {
+    return this.getBasePathProxy().basePath;
+  }
+
+  get targetPort() {
+    return this.getBasePathProxy().targetPort;
+  }
+
+  public async configure({
+    blockUntil,
+    isKibanaPath,
+  }: Pick<BasePathProxyServerOptions, 'blockUntil' | 'isKibanaPath'>) {
+    const [devConfig, httpConfig] = await Promise.all([
+      k$(this.configService.atPath('dev', DevConfig))(first(), toPromise()),
+      k$(this.configService.atPath('server', HttpConfig))(first(), toPromise()),
+    ]);
+
+    this.basePathProxy = new BasePathProxyServer(this.logger.get('server'), {
+      blockUntil,
+      devConfig,
+      httpConfig,
+      isKibanaPath,
```

Comment:
object-literal-sort-keys, lol

Reply:
? (or you mean that `object-literal-sort-keys` formed almost correct sentence? :))

Reply:
It's just funny that blockUntil and isKibanaPath are not consecutive, they go together. but it's because of sorting.

Reply:
Ah, right, `object-literal-sort-keys` isn't ideal :)